Terraform.gitignore

```diff
@@ -35,3 +35,22 @@ override.tf.json
 # Ignore CLI configuration files
 .terraformrc
 terraform.rc
+
+# Ignore hcl files
+.terraform.lock.hcl
+*.terraform.*
+
+# Ignore graph files
+*.dot
+
+# to skip MacOS custom attributes. It appears when creating a .bicep file.
+.DS_Store
+
+# Ignore plan files (optional) as some might save plan before destroy the terraform configuration
+planout
+
+# Ignore tfplan files, saved during terraform plan execution
+*.out
```

Comment:
Given the naming of the plan files is entirely user-driven I don't think we can assume that every user of terraform will name their plan outputs $something.out